packages/itwin/tree-widget/src/components/trees/experimental/common/UseHierarchyVisibility.ts

```diff
@@ -0,0 +1,113 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MutableRefObject, useEffect, useRef, useState } from "react";
+import { defer, distinct, mergeMap, Subject, takeUntil } from "rxjs";
+import { BeEvent } from "@itwin/core-bentley";
+import { HierarchyNode } from "@itwin/presentation-hierarchies";
+import { PresentationHierarchyNode } from "@itwin/presentation-hierarchies-react";
+import { VisibilityStatus } from "../../VisibilityTreeEventHandler";
+
+interface HierarchyVisibilityHandler {
+  getVisibilityStatus: (node: HierarchyNode) => Promise<VisibilityStatus>;
+  changeVisibility: (node: HierarchyNode, on: boolean) => Promise<void>;
+  onVisibilityChange: BeEvent<() => void>;
+  dispose: () => void;
+}
+
+interface UseNodesVisibilityProps {
```

Comment:
should be `UseHierarchyVisibilityProps`?

Reply:
Renamed